Formula/nlopt.rb

```diff
@@ -13,81 +13,135 @@ class Nlopt < Formula
     sha256 "e3ea4064beff9f39ab2fed6a9e5f643a18fd7abb2b2302713ce7f58f970ceb09" => :sierra
   end
 
-  head do
-    url "https://github.com/stevengj/nlopt.git"
-    depends_on "cmake" => :build
-    depends_on "swig" => :build
-  end
-
-  depends_on "numpy"
+  depends_on "cmake" => [:build, :test]
 
   def install
-    ENV.deparallelize
+    args = *std_cmake_args + %w[
+      -DNLOPT_GUILE=OFF
+      -DNLOPT_MATLAB=OFF
+      -DNLOPT_OCTAVE=OFF
+      -DNLOPT_PYTHON=OFF
+      -DNLOPT_SWIG=OFF
+      -DNLOPT_TESTS=OFF
+    ]
 
-    if build.head?
-      system "cmake", ".", *std_cmake_args,
-                      "-DBUILD_MATLAB=OFF",
-                      "-DBUILD_OCTAVE=OFF",
-                      "-DWITH_CXX=ON"
-    else
-      system "./configure", "--prefix=#{prefix}",
-                            "--enable-shared",
-                            "--with-cxx",
-                            "--without-octave"
+    mkdir "build" do
+      system "cmake", *args, ".."
       system "make"
-    end
-    system "make", "install"
-
-    # Create library links for C programs
-    %w[0.dylib dylib a].each do |suffix|
-      lib.install_symlink "#{lib}/libnlopt_cxx.#{suffix}" => "#{lib}/libnlopt.#{suffix}"
+      system "make", "install"
```

Comment:
To reduce the maintenance burden, we should copy examples from https://github.com/stevengj/nlopt/tree/master/test via `pkgshare.install "test/<file>"` and then use these in the tests. `t_tutorial.cxx` appears to be the best candidate. Optionally we could also try `box.c` or `lorentzfit.c` but neither appear to be tested by upstream. The test code below, while thorough, may break due to API changes, or build system changes in future releases. It would be nice to let upstream provide and maintain the tests.

Reply:
Looks like `t_tutorial.cxx` was added after the release was tagged, so looks like it is going to have to be `box.c` or `lorentzfit.c` for now.